src/dawn/Optimizer/PassSetBoundaryCondition.cpp

```diff
@@ -279,8 +286,9 @@ bool PassSetBoundaryCondition::run(const std::shared_ptr<StencilInstantiation>&
                 stmt->accept(visitor);
                 std::vector<std::shared_ptr<Stmt>> stencilCallWithBC_;
                 stencilCallWithBC_.emplace_back(IDtoBCpair->second);
-                stencilCallWithBC_.emplace_back(stencilcallstmt->second);
-
+                if(visitor.getStencilCallsToReplace().size() > 0) {
+                  stencilCallWithBC_.emplace_back(visitor.getStencilCallsToReplace()[0]);
```

Comment:
can it be there are more than 1 elements? It does not look like, so how about using simply a srd::shared_ptr that is null if not found? instead of a vector?

Reply:
upon testing, you seem to be right but I wonder why.
If we have a stencil like
```
stencil multiplecalls {
  storage a, b;

  boundary_condition(zero(), b);
  boundary_condition(zero(), a);

  void Do() {
    bar(a, b);
    if(global_flag) {
      bar(a, b);
    } else {
      foo(a, b);
    }
    if(global_flag) {
      bar(a, b);
    } else {
      foo(a, b);
    }
  }
};
```
we generate a total of 5 stencils (each only called once). Why is that?